### PR TITLE
Configure S3 bucket for backups with offsite replication

### PR DESCRIPTION
**Pull request by cat5inthecradle**
**TL;DR:** This creates an S3 Bucket in Prod that gets automatically replicated to our offsite account. This bucket can be used to store one-off and automated backups.

This PR creates a bucket, `cdo-backup` in our primary AWS account and region (us-east-1). This bucket is configured to replicate all objects to an S3 bucket in our offsite AWS account, in a different region.

The destination bucket, `cdo-backup-offsite` is also defined in a new template. This can be manually launched in the offsite account to create the destination bucket and associated permissions.

This will give us an S3 bucket for backups, with object versioning, that is replicated to a different region automatically. The only shared information between the AWS accounts is that Prod needs to know the name of the target bucket, and Offsite needs to know the ARN of the Role S3 will be assuming for replication.

The bucket `cdo-backups` (plural) exists, but looks old. Let’s let it be and migrate files if we choose.

## Links

- [AWS S3 Replication](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication.html)
- [AWS S3 Storage Classes](https://aws.amazon.com/s3/storage-classes/)

## Testing story

Deploying this does not affect production resources, nor is it released via DTP. It could be iterated upon until successful.

## Deployment strategy

1. Deploy iam stack
2. Deploy offsite stack - _The offsite AWS stack needs to be deployed to us-west-2 (or anything not us-east-1)_
3. Deploy s3_buckets stack

## Follow-up work

- Configure appropriate [replication metric](https://docs.aws.amazon.com/AmazonS3/latest/userguide/replication-metrics.html) alarms
- Alerting if replication fails
- Possible adjustments to target bucket storage class
- Plan to migrate content from cdo-backups
- Update documentation around backups and the offsite account.

## Privacy

- Both S3 buckets have public access disabled.
- _what_ goes into these buckets requires separate discussions.
- how backups related to policies around data deletion is a separate discussion. Deletes are NOT persisted with the current configuration, in an effort to preserve data in the case of accidental deletion.

## Security

- See Draft technical proposal around securing access to the offsite AWS account (ask Darin for link)

## PR Checklist:

- [ ] ~Tests provide adequate coverage~
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [ ] ~New features are translatable or updates will not break translations~
- [ ] Relevant documentation has been added or updated
- [ ] ~User impact is well-understood and desirable~
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: aws/cloudformation/offsite.template.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: "Offsite CDO Resources"
+
+Parameters:
+  OnsiteProdBackupRoleArn:
+    Type: String
+
+Resources:
+  OffsiteBackupBucket:
```
Comment (cat5inthecradle) [inline aws/cloudformation/offsite.template.yml]: These should go in a StackSet where we specify the region

**Comment by cat5inthecradle** [comment]
Checking in on this old PR. Still valid and ready for review.

**Comment by sureshc** [comment]
I can look this week! Likely tomorrow afternoon.
